arch/xtensa/core/xtensa-asm2-util.S

```diff
@@ -174,7 +174,7 @@ _restore_context:
 	l32i a0, a1, ___xtensa_irq_bsa_t_scompare1_OFFSET
 	wsr a0, SCOMPARE1
 #endif
-#if XCHAL_HAVE_THREADPTR && defined(CONFIG_THREAD_LOCAL_STORAGE)
+#if XCHAL_HAVE_THREADPTR
```

Comment:
This adds overhead to existing non-TLS/non-USERSPACE builds.  Not a huge deal, but the save is currently elided for THREADPTR devices that don't enable TLS (which is all of them IIRC, as we don't do TLS with xcc?), and now they have to save it even though they won't use it.

Reply:
yep ... xcc does not have TLS. TLS was the intial attempt, similar to RISCV. I can add a `CONFIG_USERSPACE` condition there though.

Reply:
Just added kept this condition and added an OR for `defined(CONFIG_USERSPACE)`, so there is no overhead.